examples/games/flappy_bevy.rs

```diff
@@ -0,0 +1,278 @@
+    .add_event::<SpawnBird>();
+
+    app.run()
+}
+
+pub struct Art {
+    bird_icon: Handle<Image>,
+}
+
+fn load_art(mut commands: Commands, asset_server: Res<AssetServer>) {
+    commands.insert_resource(Art {
+        bird_icon: asset_server.load("branding/icon.png"),
+    });
+}
+struct GenerateChunk {
+    new_chunk_index: i32,
+}
+
+fn advance_camera(
+    mut q: Query<&mut Transform, With<Camera>>,
+    time: Res<Time>,
+    mut outgoing_generate_chunk_events: EventWriter<GenerateChunk>,
+) {
+    if let Ok(mut transform) = q.get_single_mut() {
+        let chunk_index = (transform.translation.x / CHUNK_SIZE).floor() as i32;
+        transform.translation.x += time.delta().as_secs_f32() * CAMERA_SPEED;
+        let new_chunk_index = (transform.translation.x / CHUNK_SIZE).floor() as i32;
+        if chunk_index != new_chunk_index {
+            outgoing_generate_chunk_events.send(GenerateChunk {
+                new_chunk_index: new_chunk_index + 3,
+            });
+        }
+    }
+}
+
+#[derive(Component)]
+struct Obstacle;
+
+fn terrain_cleanup(
+    mut commands: Commands,
+    q: Query<(Entity, &Transform), With<Obstacle>>,
+    cam: Query<&Transform, With<Camera>>,
+) {
+    let cam_x = cam.single().translation.x;
+    for (e, t) in q.iter() {
+        if t.translation.x < cam_x - CLEANUP_X_DIST {
+            commands.entity(e).despawn();
+        }
+    }
+}
+
+fn terrain_gen(
+    mut commands: Commands,
+    mut incoming_generate_chunk_events: EventReader<GenerateChunk>,
+) {
+    for ev in incoming_generate_chunk_events.iter() {
+        let x_pos = CHUNK_SIZE * ev.new_chunk_index as f32;
+        // generate some terrain within x_pos..x_pos+width
+        let gap_y_pos = GAME_HEIGHT * (randf() - 0.5);
+        let pillar_width = 50.0 + 110.0 * randf();
+        // make the gap no narrower than the pillar is wide
+        let gap_size = (50.0 + 250.0 * randf()).max(pillar_width);
+        for (top_y_pos, bottom_y_pos) in [
+            (-SCREEN_HEIGHT * 0.5, gap_y_pos - gap_size * 0.5),
+            (gap_y_pos + gap_size * 0.5, SCREEN_HEIGHT * 0.5),
+        ] {
+            let pillar_origin = Vec2::new(x_pos, (top_y_pos + bottom_y_pos) * 0.5);
+            let pillar_size = Vec2::new(pillar_width, bottom_y_pos - top_y_pos);
+            commands
+                .spawn_bundle(SpriteBundle {
+                    sprite: Sprite {
+                        color: Color::rgb(0.25, 0.25, 0.75),
+                        custom_size: Some(pillar_size),
+                        ..default()
+                    },
+                    transform: Transform::from_translation(pillar_origin.extend(0.0)),
+                    ..default()
+                })
+                .insert(Obstacle);
+        }
+    }
+}
+
+fn spawn_camera(mut commands: Commands) {
+    commands.spawn_bundle(Camera2dBundle::default());
+}
+
+#[derive(Component, Default)]
+struct Velocity {
+    velocity: Vec2,
+}
+
+#[derive(Component)]
+struct Chaos;
```

Comment:
I don't think that Chaos, Gravity or DriftToCenter components have value; the behavior can be configured based on adding and removing the systems.

Reply:
You can't remove systems from a schedule, right? At most have a run criterion to disable them. Also these components are not global, but specific to the properties of specific entities it seems, so globally adding or removing systems wouldn't work.

Reply:
These systems could all operate on Bird. I've been using a Component for each distinct behavior if possible for clarity.

Reply:
Yep, the main thing here is that when designing which components I use, I try and think about this in terms of future extensibility.

I think that Gravity is plausible to extend, but the others probably aren't? That said, I think there's an argument for keeping this as minimal as possible to improve the clarity as a learning example.

Reply:
I would argue for keeping these micro-components separate as they can be copy-pasted into new users' projects.